### PR TITLE
Improvement in RipUtils

### PR DESCRIPTION
**Pull request by Gaboso**
Defined a constant instead of duplicating this literal "imgur.com".
Defined a constant instead of duplicating this literal "http: //".
Moved strings to the left side of the string comparison.
Added a private constructor to hide the implicit public one.
Removed redundant initializers.
Changed indexOf(char) to single quotes for optimized for single characters.
Removed commented code blocks.
In the urlFromDirectoryName method, the first 'if' is always null, so I removed it.
Changed constant logger name to LOGGER.
Replace usage of System.out by Logger.

### CODE REVIEW

**Comment by rautamiekka** [comment]
While it's not 100% necessary, escaping the dots in website addresses in a RegEx ensures 100% success and is a performance optimization.

Also, just in case a site switches from HTTP to HTTPS or the other way around later a constant should be defined (instead of hardcoding in the code) and used like this (imitated from the very code) in Imgur's case:
```java
private static final String SITE_PROTOCOL = "http";
private static final String SITE_ADDRESS = SITE_PROTOCOL + "://imgur.com";

some_code(SITE_ADDRESS + "/01.png");
```

File: src/main/java/com/rarchives/ripme/utils/RipUtils.java [hunk 1]
```diff
@@ -9,97 +22,91 @@
...
-public class RipUtils {
-    private static final Logger logger = Logger.getLogger(RipUtils.class);
+    private RipUtils() {
+    }
 
     public static List<URL> getFilesFromURL(URL url) {
         List<URL> result = new ArrayList<URL>();
 
-        logger.debug("Checking " + url);
+        LOGGER.debug("Checking " + url);
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/utils/RipUtils.java]: Lowercase logger is used throughout ripme. Sure it's a convention for "constants" to be in all caps, but that static final variable isn't even benefitting from pretending to be a constant. Sure you can't reassign it, but as an output logger, it's state changes. I don't think anything useful is documented there. I'd prefer keeping it lowercase `logger` for consistency with the rest of the program and to reduce the size of unnecessary diffs. If you're serious about this change, it needs to change in the whole project.

File: src/main/java/com/rarchives/ripme/utils/RipUtils.java [hunk 2]
```diff
@@ -1,5 +1,18 @@
...
+import com.rarchives.ripme.ripper.rippers.ImgurRipper;
+import com.rarchives.ripme.ripper.rippers.ImgurRipper.ImgurAlbum;
+import com.rarchives.ripme.ripper.rippers.ImgurRipper.ImgurImage;
+import com.rarchives.ripme.ripper.rippers.VidbleRipper;
+import com.rarchives.ripme.ripper.rippers.video.GfycatRipper;
+import org.apache.commons.lang.math.NumberUtils;
+import org.apache.log4j.Logger;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/utils/RipUtils.java]: Just wondering for future consistency, what tool or scheme did you use to sort the imports?

File: src/main/java/com/rarchives/ripme/utils/RipUtils.java [hunk 3]
```diff
@@ -9,97 +22,91 @@
...
 
     public static List<URL> getFilesFromURL(URL url) {
         List<URL> result = new ArrayList<URL>();
 
-        logger.debug("Checking " + url);
+        LOGGER.debug("Checking " + url);
         // Imgur album
-        if ((url.getHost().endsWith("imgur.com"))
-                && url.toExternalForm().contains("imgur.com/a/")) {
+        if ((url.getHost().endsWith(IMGUR_COM)) && url.toExternalForm().contains("imgur.com/a/")) {
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/utils/RipUtils.java]: Not going to replace the `imgur.com` at the end of this line?

File: src/main/java/com/rarchives/ripme/utils/RipUtils.java [hunk 4]
```diff
@@ -109,38 +116,31 @@ else if (url.toExternalForm().contains("eroshare.com")) {
...
                 return result;
             } catch (MalformedURLException e) {
-                logger.error("[!] Not a valid URL: '" + url + "'", e);
+                LOGGER.error("[!] Not a valid URL: '" + url + "'", e);
             }
         }
 
-        if (url.getHost().equals("imgur.com") ||
-                url.getHost().equals("m.imgur.com")) {
+        if (url.getHost().equals(IMGUR_COM) || "m.imgur.com".equals(url.getHost())) {
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/utils/RipUtils.java]: make a constant for M_IMGUR_COM

File: src/main/java/com/rarchives/ripme/utils/RipUtils.java [hunk 5]
```diff
@@ -149,122 +149,156 @@ public static Pattern getURLRegex() {
...
-        if (url == null) url = urlFromSiteDirectoryName(dir, "drawcrowd",   "http://drawcrowd.com/", "");
-        if (url == null) url = urlFromSiteDirectoryName(dir.replace("-", "/"), "ehentai", "http://g.e-hentai.org/g/", "");
-        if (url == null) url = urlFromSiteDirectoryName(dir, "fapproved", "http://fapproved.com/users/", "");
-        if (url == null) url = urlFromSiteDirectoryName(dir, "vinebox", "http://finebox.co/u/", "");
-        if (url == null) url = urlFromSiteDirectoryName(dir, "imgbox", "http://imgbox.com/g/", "");
-        if (url == null) url = urlFromSiteDirectoryName(dir, "modelmayhem", "http://www.modelmayhem.com/", "");
+        String url = urlFromImgurDirectoryName(dir);
+
+        if (url == null)
+            url = urlFromImagefapDirectoryName(dir);
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/utils/RipUtils.java]: So this whole thing, as a style guideline one of three things can happen. `if` statement with a single statement body all on a single line (as it was before), the same with braces around the body (my preference), or indented WITH braces added around the body.

Absolutely please do not use the indented form without adding brackets.

IMO, the single-line forms are more readable in this case.

If you want to make a change here, please add brackets around the if bodies and it's up to you whether to reformat so the bodies are all on their own lines.

Also looking at this code now, I feel like there's got to be a better way of writing this bit of code (more efficient, etc.). If you make that sort of change, please do it is in a separate commit or pull request.

File: src/main/java/com/rarchives/ripme/utils/RipUtils.java [hunk 6]
```diff
@@ -149,122 +149,156 @@ public static Pattern getURLRegex() {
...
+
         //if (url == null) url = urlFromSiteDirectoryName(dir, "8muses",      "http://www.8muses.com/index/category/", "");
         return url;
     }
 
     private static String urlFromSiteDirectoryName(String dir, String site, String before, String after) {
-        if (!dir.startsWith(site + "_")) {
+        if (!dir.startsWith(site + "_"))
             return null;
-        }
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/utils/RipUtils.java]: Don't remove braces around if or loop bodies. They should be required AFAIC.

File: src/main/java/com/rarchives/ripme/utils/RipUtils.java [hunk 7]
```diff
@@ -149,122 +149,156 @@ public static Pattern getURLRegex() {
...
+
+        String directory = dir.substring((site + "_").length());
+        return before + directory + after;
     }
 
     private static String urlFromRedditDirectoryName(String dir) {
-        if (!dir.startsWith("reddit_")) {
+        if (!dir.startsWith("reddit_"))
             return null;
-        }
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/utils/RipUtils.java]: Keep braces.

File: src/main/java/com/rarchives/ripme/utils/RipUtils.java [hunk 8]
```diff
@@ -149,122 +149,156 @@ public static Pattern getURLRegex() {
...
-        else {
-            url = "http://www.imagefap.com/gallery.php?pgid=" + dir;
-        }
+
+        String url;
+        String directory = dir.substring("imagefap_".length());
+        if (NumberUtils.isDigits(directory))
+            url = "http://www.imagefap.com/gallery.php?gid=" + directory;
+        else
+            url = "http://www.imagefap.com/gallery.php?pgid=" + directory;
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/utils/RipUtils.java]: Please keep braces.

**Review by metaprime — CHANGES_REQUESTED** [review]

File: src/main/java/com/rarchives/ripme/utils/RipUtils.java (hunk 2 above)
Comment (Gaboso) [inline src/main/java/com/rarchives/ripme/utils/RipUtils.java]: I'm using IntelliJ IDEA

**Comment by metaprime** [comment]
@Gaboso Why did you close the PR? Apologies for not following up on this.

**Pull request closed without merging**